### PR TITLE
SQL-2101: Add analysis to track when vuln was first reported

### PR DESCRIPTION
**Pull request by nbagnard**
See a successful patch here : https://spruce.mongodb.com/task/mongosql_odbc_driver_code_quality_security_sbom_patch_61b2db964203a2fee9b949b407e7de51337b41d1_6679d7da503ce80007345511_24_06_24_20_32_38/files?execution=0

The SBOM now has a "analysis" section for vulns not yet fixed

### CODE REVIEW

**Review by terakilobyte — APPROVED** [review]
answered in slack, date is date first discovered which we can't dynamically generate!

File: GHSA-4grx-2x9w-596c.analysis.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"firstIssued": "2024-06-05T15:32:43Z","state": "under_investigation","detail": "No fixed upgrade is available!"}
```
Comment (jchemburkar) [inline GHSA-4grx-2x9w-596c.analysis.json:1]: should we put these in somewhere centralized like a subfolder of `resources`?

**Review by jchemburkar — APPROVED** [review]
lgtm, just one thought. Are we going to maintain these indefinitely as they come up / remove them as we resolve?

File: GHSA-4grx-2x9w-596c.analysis.json (hunk 1 above)
Comment (nbagnard) [reply GHSA-4grx-2x9w-596c.analysis.json]: Yes, definitely!
I added the change.

**Pull request merged**